nsapass

```diff
@@ -401,18 +416,24 @@ try:
     elif args.command == 'get':
         nsa_get(args.name, args.index)
     elif args.command == 'add':
+        lock_db()
         nsa_add(args.name, args.passwd, args.note)
     elif args.command == 'gen':
+        lock_db()
```

Comment:
this means user won't be able to run, say, `nsapass add ...` while another instance of `nsapass gen ...` is running in the early stage of confirming selection of generated password.  

e.g. say the user has `nsapass gen  nick 25 printable` running, with a password presented to him to say whether he likes it.  here the process can be blocked indefinitely as it waits for the user to finally decides whether he likes the generated password.  during this time, the user may execute, say, `nsapass mod anothernick 0 -p newpassword` in another shell to edit a previous password manually, but it won't work, even tho `nsapass gen...` still doesn't need a lock.

do you think we should give up this case?

i can think of the following, but not sure what you think about it.  basically to lock inside the `load` function, and indicate desire to locking by a global variable (similar to how `SHOW_PASSWORD` works).  basically:

1. instead of calling `lock_db()` here (and in other places in the dispatcher), we say `lock = True`.
2. we move the content of the `lock_db` function inside `load` function under an `if lock: ...` block.
3. and since no other function is using the locking code in this case, the locking code will not be a function but rather a part of `load`.  so, no decorations either.

this way the `nsapass gen ...` restriction above would be removed as locking happens after calling `load()`, which is the step after the user had decided that he is happy with the generated password.

is there any case against this?

(thanks a lot for your time and sorry for making you read all this :cry:)

Reply:
As you may have realized, I really don't like global variables very much. But we can make locking the responsibility of `load` (thus delaying taking the lock until that point), which is what I think you want by giving it an extra parameter that indicates whether a lock should be taken or not... and at the moment at least, it is clear for every call to `load()` whether the lock should be taken or not. (See the update I just pushed.)

Reply:
you're right.  not worth the complexity.  i'll merge this.

ty for your help + time + efforts. highly appreciated.  and apologies for any pointless inconvenience.